### PR TITLE
Add support for Google Pay payment method

### PR DESCRIPTION
**Pull request by hpariwat**
#### 1. Objective

Add support for Google Pay payment method. Allow customers to pay using cards that are connected to their Google Account.

#### 2. Description of change

* Add an option on Admin dashboard to allow customers to pay with Google Pay.
* Add Google-related configurable options such as Merchant ID, Request Billing Address, and Request Phone Number.
* Make it possible to fetch card brands from Capability.
* Google Pay uses card's existing command pool.

#### 3. Quality assurance

**🔧 Environments:**

- **Platform version**: Magento CE 2.2.3
- **Omise plugin version**: Omise-Magento 2.24.0
- **PHP version**: 7.1.15

**✏️ Details:**

A new section on the Admin configuration site.
Screen Shot 2565-05-27 at 10 15 11

The Merchant ID, Request Billing Address, and Request Phone Number options are sent directly upon connecting with Google server, depending on business needs.

If Request Billing Address is not enabled, then Request Phone Number will not be visible.

Screen Shot 2565-05-27 at 17 15 10

A Google Pay button will appear if it's enabled on the Admin's site.
Screen Shot 2565-05-27 at 09 49 39

Upon readiness check failure, this error message will appear.
Screen Shot 2565-05-27 at 09 51 33

* Testing checklist: with live/test account, and with 3DS/non-3DS enabled.

#### 4. Impact of the change

Customers can now pay with Google Pay.

#### 5. Priority of change

Normal.

#### 6. Additional Notes

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -663,6 +663,57 @@
...
+                        <comment><![CDATA[The merchant ID will be available after registering with the <a href="https://pay.google.com/business/console">Google Pay Business Console</a>.]]></comment>
+                        <config_path>payment/omise_cc_googlepay/merchant_id</config_path>
+                    </field>
+                    <field id="request_billing_address" translate="label comment" type="select" sortOrder="508" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Request Billing Address</label>
+                        <comment>Request customer's name and billing address from their Google Account upon checkout.</comment>
+                        <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                        <config_path>payment/omise_cc_googlepay/request_billing_address</config_path>
+                    </field>
+                    <field id="request_phone_number" translate="label comment" type="select" sortOrder="509" showInDefault="1" showInWebsite="1" showInStore="1">
```
Comment (kan98) [inline etc/adminhtml/system.xml:686]: we will only fetch the phone number if billing address is also enabled right? is there any way to dynamically disable this if billing is disabled? or maybe a good idea to add in the description about this

File: Controller/Callback/Threedsecure.php [hunk 2]
```diff
@@ -82,7 +83,10 @@ public function execute()
             return $this->redirect(self::PATH_CART);
         }
 
-        if ($payment->getMethod() !== 'omise' && $payment->getMethod() !== 'omise_cc') {
+        if ($payment->getMethod() !== 'omise'
+            && $payment->getMethod() !== 'omise_cc'
+            && $payment->getMethod() !== CcGooglePay::CODE
```
[the remark this answers is not in the document]
Comment (hpariwat) [reply Controller/Callback/Threedsecure.php]: Fixed here: [e5ea073](https://github.com/omise/omise-magento/pull/351/commits/e5ea073bc006540b7ad229e8dd19304279e807c5)

File: etc/adminhtml/system.xml (hunk 1 above)
Comment (hpariwat) [reply etc/adminhtml/system.xml:686]: Nice idea! Added fix here [e5ea073](https://github.com/omise/omise-magento/pull/351/commits/e5ea073bc006540b7ad229e8dd19304279e807c5) and now it's not visible if billing address is not requested.

Screen Shot 2565-05-27 at 17 15 10

**Review by kan98 — APPROVED** [review]

**Review by new4762 — APPROVED** [review]

**Review by aashishgurung — APPROVED** [review]
LGTM

**Review by tookaa2 — APPROVED** [review]

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -663,6 +663,60 @@
...
+                    <comment>Enable customers to pay using credit/debit cards connected to their Google account. </comment>
+                    <field id="active" translate="label comment" type="select" sortOrder="506" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Enable/Disable</label>
+                        <comment>Enable Google Pay payment method.</comment>
+                        <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                        <config_path>payment/omise_cc_googlepay/active</config_path>
+                    </field>
+                    <field id="merchant_id" translate="label" type="text" sortOrder="507" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Google Pay Merchant ID</label>
+                        <comment><![CDATA[The merchant ID will be available after registering with the <a href="https://pay.google.com/business/console">Google Pay Business Console</a>.]]></comment>
```
Comment (kan98) [inline etc/adminhtml/system.xml:677]: can you note that this is not needed for test mode?

**Review by som-m — APPROVED** [review]
- [x] 3ds payments - auto capture
- [x] 3ds payments - manual capture
- [x] non 3ds payments - auto capture
- [x] non 3ds payments - manual capture
- [x] multi currency payments
- [x] capture payments
- [x] fetch available card brands
image

File: etc/adminhtml/system.xml (hunk 3 above)
Comment (hpariwat) [reply etc/adminhtml/system.xml]: Added here: [08e206d](https://github.com/omise/omise-magento/pull/351/commits/08e206dc0a2767c061f61a2d64f65744b56388c2)

**Pull request merged**